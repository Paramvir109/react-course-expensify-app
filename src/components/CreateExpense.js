import React from 'react'
import { connect } from 'react-redux';
import {addExpense} from './../actions/expenses'
import ExpenseForm from './ExpenseForm';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <h1>This is from my add expense component</h1>
        <ExpenseForm onSubmit={this.onSubmit}/>
      </div>
    );
  }
}

// export const AddExpensePage = (props) => (
//     <div>
//       <h1>This is from my add expense component</h1>
//       <ExpenseForm 
//         onSubmit={(expense) => {
//           // props.dispatch(addExpense(expense)); will change it to make test-able as well
//           props.onSubmit(expense);
//           props.history.push('/');//Will automatically redirect to dashboard page
//         }}
        
//       />
//     </div>
// );

//optional -> used to dispatch actions
const mapDispatchToProps = (dispatch) => {
  return {
    addExpense : (expense) => dispatch(addExpense(expense))
  }
}

export default connect(undefined, mapDispatchToProps)(AddExpensePage);