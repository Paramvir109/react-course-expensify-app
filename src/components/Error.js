import { Link } from 'react-router-dom'
import React from 'react'

 //Link tag causes it to rerender(Not a full page refresh)
 const NotFoundPage = () => (
    <div>
      404! - <Link to="/">Go home</Link>
    </div>
);
  export default NotFoundPage;