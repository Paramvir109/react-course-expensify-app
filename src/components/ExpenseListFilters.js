import React from 'react'
import { connect } from 'react-redux';
import { setTextFilter,sortByAmount,sortByDate, setEndDate, setStartDate} from './../actions/filters';
import 'react-dates/initialize';
import uuid from 'uuid'

import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused : null
    }
    onDatesChange = ({startDate, endDate}) => {
        this.props.setEndDate(endDate);
        this.props.setStartDate(startDate);

    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    onSortChange = (e) => {
        this.props.filters.sortBy === 'date'? this.props.sortByAmount() : this.props.sortByDate()
    }

    render() {
       return (
            <div>
                <input 
                    type="text" 
                    value={this.props.filters.text} 
                    onChange={this.onTextChange}>
                </input>
                <select 
                    value={this.props.filters.sortBy} 
                    onChange={this.onSortChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    startDateId={uuid()}
                    endDateId={uuid()}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    showClearDates={true}
                />
            </div>
        );
    }
}

//props above contains two things 1.) filters 2.) dispatch method(bydefault)
const mapStateToProps = (state) => {
    return {
        filters : state.filters
    }
}
const mapDispatchToProps = (dispatch) => {
    return {
      setTextFilter : (value) => dispatch(setTextFilter(value)),
      setStartDate : (startDate) => dispatch(setStartDate(startDate)),
      setEndDate : (endDate) => dispatch(setTextFilter(endDate)),
      sortByAmount : () => dispatch(sortByAmount()),
      sortByDate : () => dispatch(sortByDate()),
    }
  }

export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);