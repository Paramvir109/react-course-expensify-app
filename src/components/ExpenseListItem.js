import React from 'react'
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

//dispatch method available from connect
export const ExpenseListItem = ({description, createdAt, amount, id, dispatch}) => (
    <div>
    <NavLink activeClassName="is-active" to={"/edit/" + id}>
         <h3>{description}</h3>
    </NavLink>
        <p>{amount} - {createdAt}</p>
       
    </div>
);



export default connect()(ExpenseListItem);