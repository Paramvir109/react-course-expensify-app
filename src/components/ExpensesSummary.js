import React from 'react'
import { connect } from 'react-redux';
import selectExpenseList from './../selectors/expenses';
import expensesTotal from './../selectors/expenses-total';
import numeral from 'numeral';



export const ExpensesSummary = (props) => {
    const expenseWord = props.expenseCount === 1 ? 'expense' : 'expenses';
    return(

        <div>
            <h1>
                Viewing {props.expenseCount} {expenseWord} totalling  Rs.{numeral(props.expensesSum/100).format('0,0.00')} 
            </h1>
        </div>
    )

} 

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenseList(state.expenses, state.filters)
    return {
        expenseCount : visibleExpenses.length,
        expensesSum : expensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);