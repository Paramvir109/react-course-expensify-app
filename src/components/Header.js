import { NavLink } from 'react-router-dom'
import React from 'react'
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth'

//We use navlink to provide more props which helps in styling
export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
        <NavLink  activeClassName="is-active" to="/dashboard">Dashboard</NavLink>
        <NavLink activeClassName="is-active" to="/create">Create Expense</NavLink>
        <button onClick={startLogout}>Log out</button>
    </header>
);
const mapDispatchToProps = (dispatch) => ({
    startLogout : () => dispatch(startLogout())
})
export default connect(undefined,mapDispatchToProps)(Header);