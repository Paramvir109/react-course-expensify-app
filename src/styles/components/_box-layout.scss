.box-layout {
    align-items: center;
    background-image: url('/images/bg.jpg');//only public folder is accessible in web browser
    background-size: cover;//to still make sures it covers the container
    height: 100vh;
    display: flex;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(white,  .25);
    padding: $l-size $m-size;
    border-radius: 3px;
    text-align: center;
    width: 25rem;
}
.box-layout__title {
    line-height: 1;
    margin:0 0 $m-size 0;
}