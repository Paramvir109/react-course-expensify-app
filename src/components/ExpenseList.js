import React from 'react'
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenseList from './../selectors/expenses';
export const ExpenseList = (props) => (//Route sends some built in props
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        <div className="list-body"> 
            {props.expenses.length === 0 ? (<div className="list-item list-item__message"><span>No expenses</span></div>):
                (props.expenses.map((expense) => (<ExpenseListItem key={expense.id} {...expense}/>)))
            }
        </div>
        
    </div>
);
//return value of connect is the function which is to be called to create a hoc

// const connectedExpenseList = connect((state) => {
//     //here this function tells bout which properties of store can be accesssed by the component(connectedExpenseList)
//     return {
//         expenses : state.expenses
//     }

// })(ExpenseList);

// export default connectedExpenseList;

//OR
const mapStateToProps = (state) => {
    return {
        expenses : selectExpenseList(state.expenses, state.filters)
    }

}
export default connect(mapStateToProps)(ExpenseList);

