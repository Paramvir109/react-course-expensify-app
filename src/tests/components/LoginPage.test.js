import React from 'react';
import { shallow } from 'enzyme';
import {LoginPage} from '../../components/LoginPage';

let startLogin, wrapper;
beforeEach(() => {
    startLogin = jest.fn()
     
    wrapper =  shallow(<LoginPage startLogin={startLogin}/>);
})

test('should match snapshot of login page', () => {
    expect(wrapper).toMatchSnapshot();
})
test('should call logout fn', () => {
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
})
