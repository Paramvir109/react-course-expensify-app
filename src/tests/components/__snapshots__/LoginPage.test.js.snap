// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot of login page 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
